### PR TITLE
Apply flaky test settings for testJava[9|12]

### PR DESCRIPTION
**Pull request by minwoox**
I realized that `project.ext.configureFlakyTests` wasn't applied for `:core:testJava12`, thus `InvalidPathWithDataTest` still fails

### CODE REVIEW

**Review by trustin — APPROVED** [review]
Thanks, @minwoox!

**Review by jrhee17 — APPROVED** [review]
👍 👍 👍

File: core/build.gradle [hunk 1]
```diff
@@ -31,19 +31,22 @@ mrJarVersions.each { version->
...
     }
 
-    task "testJava${version}"(type: Test, group: 'Verification', description: "Runs unit tests for Java ${version} source set") {
+    tasks.register("testJava${version}", Test) {
+        group = 'Verification'
+        description = "Runs unit tests for Java ${version} source set"
         testClassesDirs = sourceSets."java${version}Test".output.classesDirs
         classpath = sourceSets."java${version}Test".runtimeClasspath
 
+        project.ext.configureFlakyTests(it)
```
Comment (jrhee17) [inline core/build.gradle:40]: optional; I haven't tried this myself, but I'm wondering if there's a reason not to merge `configureFlakyTests` into `configureCommonTestSettings` now that all test tasks are using the junit api regardless of the engine

Comment (minwoox) [reply core/build.gradle:40]: I think it's a good idea. Let me try that. 👍

Comment (minwoox) [reply core/build.gradle]: Addressed. 😉 
Will send a PR to line/gradle-scripts as well.

**Review by ikhoon — APPROVED** [review]

**Pull request merged**